OMEdit/OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp

```diff
@@ -2838,34 +2907,8 @@ void GraphicsView::mouseDoubleClickEvent(QMouseEvent *event)
       pShapeAnnotation->showShapeAttributes();
       return;
     }
-  }
-  // handle double click on the Component.
-  QGraphicsItem *pGraphicsItem = itemAt(event->pos());
-  if (pGraphicsItem && pGraphicsItem->parentItem()) {
-    Component *pComponent = dynamic_cast<Component*>(pGraphicsItem->parentItem());
-    if (pComponent) {
-      Component *pRootComponent = pComponent->getRootParentComponent();
-      if (pRootComponent) {
-        if (mpModelWidget->getLibraryTreeItem()->getLibraryType() == LibraryTreeItem::CompositeModel) {
-          pRootComponent->showSubModelAttributes();
-          return;
-        } else if (mpModelWidget->getLibraryTreeItem()->getLibraryType() == LibraryTreeItem::OMS) {
-          removeCurrentConnection();
-          pRootComponent->handleOMSComponentDoubleClick();
-          return;
-        } else {
-          removeCurrentConnection();
-          removeCurrentTransition();
-          /* ticket:4401 Open component class with shift + double click */
-          if (QApplication::keyboardModifiers() == Qt::ShiftModifier) {
-            pRootComponent->openClass();
-            return;
-          } else {
-            pRootComponent->showParameters();
-            return;
-          }
-        }
-      }
+    if (!handleDoubleClickOnComponent(event)) {
```

Comment:
This should go outside the if block otherwise double click on the components will not work.

Reply:
Think I moved it in and out last time, was a bit unsure, 

How does everything else look?